maths/segmented_sieve.py

```diff
@@ -4,7 +4,26 @@
 
 
 def sieve(n: int) -> list[int]:
-    """Segmented Sieve."""
+    """
+    Segmented Sieve.
+
+    Examples:
+    >>> sieve(8)
+    [2, 3, 5, 7]
+
+    >>> sieve(27)
+    [2, 3, 5, 7, 11, 13, 17, 19, 23]
+
+    """
+
+    if n < 0:

```

Comment:
I would be OK to combine these into a single `most be a positive integer`.

What happens on floating point input?

Reply:
@cclauss 
`print(f"{sieve(87.5) = }")`
Output
```
temp = [True] * (high - low + 1)
TypeError: can't multiply sequence by non-int of type 'float'
```

Reply:
Nice doctest to add.

Reply:
@cclauss question is off topic. For some reason, my old commits end up in pull requests. I don’t understand why this is so? Every time I make a new branch and they still get there.

Reply:
I will look in a few hours…